### ISSUE TITLE
Install of Agda 2.5.4.1 from Hackage fails

### ISSUE THREAD

**Issue by tarmouu** [agda/agda#3225]
Hi,

Failed to install Agda 2.5.4.1 with cabal, see here.

```bash
tarmo@yoneda:~$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.10.3
tarmo@yoneda:~$ cabal --version
cabal-install version 1.22.6.0
using version 1.22.5.0 of the Cabal library 
tarmo@yoneda:~$ cabal install Agda
Resolving dependencies...
Configuring Agda-2.5.4.1...
Failed to install Agda-2.5.4.1
Build log ( /home/tarmo/.cabal/logs/Agda-2.5.4.1.log ):
cabal: Error: some packages failed to install:
Agda-2.5.4.1 failed during the configure step. The exception was:
user error ('/usr/bin/ghc' exited with an error:

/tmp/cabal-tmp-7733/Agda-2.5.4.1/dist/setup/setup.hs:8:8:
Could not find module ‘System.FilePath.Find’
Perhaps you meant
System.FilePath.Windows (from filepath-1.4.0.0@filep_Ey7a1in9roBAE8bUFJ5R9m)
System.FilePath (from filepath-1.4.0.0@filep_Ey7a1in9roBAE8bUFJ5R9m)
System.FilePath.Posix (from filepath-1.4.0.0@filep_Ey7a1in9roBAE8bUFJ5R9m)
Use -v to see a list of the files searched for.
)
tarmo@yoneda:~$ more .cabal/logs/Agda-2.5.4.1.log 
tarmo@yoneda:~$ 
```

Admittedly my ghc and cabal are not the newest, but they are the newest that come with my Ubuntu  version.

However after failing to install Agda 2.5.4.1, I successfully installed Agda 2.5.4 with the same ghc and cabal.

Is this right?

Thanks.

Tarmo U

**Comment by gallais** [comment]
Seems like you're missing `filemanip` to be able to compile `Setup.hs`.
The .cabal file does list it as a dependency but it is somehow ignored?

**Comment by asr** [comment]
If you could install 2.5.4 you should can install 2.5.4.1. Which version of GHC are you using?

**Comment by tarmouu** [comment]
See above in my session snippet, it's ghc 7.10.3, admittedly outdated (but newest for Ubuntu 16.04). Still I could install 2.5.4 without any problem with the same ghc and cabal.
Note that cabal created the log file Agda-2.5.4.1.log, but left it empty.

**Comment by asr** [comment]
We support GHC 7.10.3 in both 2.5.4 and 2.5.4.1. The [Hackage Matrix](https://matrix.hackage.haskell.org/package/Agda) shows 'OK' when installing 2.5.4.1 on GHC 7.10.3.

**Comment by gallais** [comment]
Could be haskell/cabal/issues/2614: cabal fails to install `filemanip` because of a conflict,
tries to compile `Setup.hs` anyways and can't because `System.FilePath.Find` can't be
found.

I'd try to run `cabal install filemanip` first.

**Comment by asr** [comment]
It seems to be what is happening:

1. Dependency on `filemanip` was added in 2.5.4.1 (see #3128).

2. Old versions of cabal-install (e.g. version 1.22.9.0) ignore the `custom-setup` section in `Agda.cabal` and report a warning

```
Warning: Agda.cabal: Ignoring unknown section type: custom-setup
```

So, as @gallais said, you should install `filemanip` first.

**Comment by tarmouu** [comment]
Thanks @gallais and @asr, this is very useful information, I'll try!

Still, since cabal gives me no information whatsoever, it's then perhaps worth mentioning filemanip in the prerequisites here:

https://agda.readthedocs.io/en/v2.5.4.1/getting-started/prerequisites.html#prerequisites

**Comment by asr** [comment]
I'll wait your confirmation before updating the prerequisites.

**Comment by tarmouu** [comment]
Tried and is well now, after installing filemanip (cabal chose 0.3.6.3) cabal smoothly installed Agda (2.5.4.1) and it seems to work. Thanks. 

Note that the warning I got from cabal on my first attempt was not useful, as it guessed that filepath rather than filemanip was missing.

**Comment by asr** [comment]
This is really a regression. I can fix the issue in master and I'll check if I also can fix the issue in 2.5.4.1 by editing the metadata in Hackage.

**Comment by asr** [comment]
It's not possible to change the metadata in Hackage for fixing the issue in 2.5.4.1. I propose to release a new bug-fix version.
